pallets/inflation/src/lib.rs

```diff
@@ -294,11 +309,13 @@ pub mod pallet {
         ) -> DispatchResult {
             ensure_root(origin)?;
 
-            let config = Self::recalculate_inflation(next_era);
-
-            ActiveInflationConfig::<T>::put(config.clone());
+            let current_config = ActiveInflationConfig::<T>::get();
+            let mut new_config = Self::recalculate_inflation(next_era);
+            // preserve the current decay factor
+            new_config.decay_factor = current_config.decay_factor;
```

Comment:
Suggestion to pass this into the `recalculate_inflation` call, to keep it consistent with the `next_era`.